hydra-task/src/main/java/com/addthis/hydra/task/map/MapFeeder.java

```diff
@@ -103,20 +103,23 @@ public MapFeeder(StreamMapper task, TaskDataSource source, int feeders) {
         }
 
         try {
+            boolean interrupt = false;
             if (source.isEnabled()) {
                 while (fillBuffer()) {
                     if (Thread.interrupted()) {
+                        interrupt = true;
                         closeSourceIfNeeded();
                     }
                 }
             }
+            final boolean interrupted = interrupt;
```

Comment:
I'm not sure I follow. It looks like being interrupted implies someone called close, and it looks like close is only ever called on one sub-task. How is the new source getting immediately closed?